File: python/baiduads-sdk/test/balance/balance_service_test.py
"""
for test BalanceService
"""
import unittest

from baiduads.balance.balance_service import BalanceService
from baiduads.balance.model import GetBalanceInfoRequestWrapper, GetBalanceInfoRequest
from baiduads.common.model import ApiRequestHeader


class TestBalanceService(unittest.TestCase):

    def setUp(self):
        self.api = BalanceService()

    def tearDown(self):
        pass

    def test_get_balance_info(self):
        """Test case for get_balance_info

        """
        request_header = ApiRequestHeader(
            token="xxxxx",
            username="xxxxx",
            password="xxxxx",
            _spec_property_naming=True
        )

        request_body = GetBalanceInfoRequest(
            productIds=[1, 502],
            _spec_property_naming=True
        )

        get_balance_info_request_wrapper = GetBalanceInfoRequestWrapper(
            header=request_header,
            body=request_body,
            _spec_property_naming=True
        )

        response_wrapper = self.api.get_balance_info(get_balance_info_request_wrapper)

        assert response_wrapper.header.status == 0
        print(response_wrapper)


if __name__ == '__main__':
    unittest.main()

File: java/baiduads-sdk/src/test/com/baidu/dev2/api/sdk/balance/BalanceServiceTest.java
package com.baidu.dev2.api.sdk.balance;

import com.baidu.dev2.api.sdk.balance.api.BalanceService;
import com.baidu.dev2.api.sdk.balance.model.GetBalanceInfoRequest;
import com.baidu.dev2.api.sdk.balance.model.GetBalanceInfoRequestWrapper;
import com.baidu.dev2.api.sdk.balance.model.GetBalanceInfoResponseWrapper;
import com.baidu.dev2.api.sdk.common.ApiRequestHeader;
import com.baidu.dev2.api.sdk.invoke.ApiException;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

import java.util.ArrayList;
import java.util.List;

public class BalanceServiceTest {
    private BalanceService balanceService;
    private ApiRequestHeader header;

    @Before
    public void before() {
        balanceService = new BalanceService();
        header = new ApiRequestHeader();
        header.setUsername("xxxxx"); // 替换为自己的用户名
        header.setPassword("xxxxx"); // 替换为自己的密码
        header.setToken("xxxxx"); // 替换为自己的token
    }

    @Test
    public void getBalanceInfo() throws ApiException {
        GetBalanceInfoRequestWrapper requestWrapper = new GetBalanceInfoRequestWrapper();
        requestWrapper.setHeader(header);

        GetBalanceInfoRequest getBalanceInfoRequest = new GetBalanceInfoRequest();
        List<Integer> productIds = new ArrayList<>();
        productIds.add(1);
        productIds.add(502);
        getBalanceInfoRequest.setProductIds(productIds);
        requestWrapper.setBody(getBalanceInfoRequest);

        GetBalanceInfoResponseWrapper responseWrapper = balanceService.getBalanceInfo(requestWrapper);
        Assert.assertTrue(responseWrapper.getHeader().getStatus() == 0);
        System.out.println(responseWrapper);
    }
}
